### PR TITLE
Use attrs + marshmallow-annotations for entities

### PR DESCRIPTION
**Pull request by friendtocephalopods**
### Summary of Changes

Use attrs to generate python classes and marshmallow-annotations to generate schemas.

This makes it far, far less painful to add new entities.

### Tests

None modified. Refactor.

### Documentation

None modified. Refactor.

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ x ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ x ] PR includes a summary of changes. 
- [ x ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ x ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does
- [ x ] PR passes `make test`
- [ x ] I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"

### CODE REVIEW

**Comment by feng-tao** [comment]
thanks @friendtocephalopods for your contributions!. Will take a look. cc @jinhyukchang as well

**Comment by jinhyukchang** [comment]
@friendtocephalopods I am not familiar with `marshmallow-annotations` and could you explain the benefit of it?

cc: @bolkedebruin @verdan

**Comment by friendtocephalopods** [comment]
Sure! Marshmallow-annotations is able to generate schemas using type hints, so you don't need to define schemas by hand.

In this case, the schemas would be useful for the common amundsen-models/amundsen-entities package that's forthcoming, as well as a PUT/POST tables endpoint for metadatalibrary I'd be happy to create a separate PR for

**Comment by bolkedebruin** [comment]
O thats interesting! It could be useful for airflow as well if so we should put this in a separate package maybe (ie the schema definitions). The meta stuff attracts me less ;-)

**Comment by verdan** [comment]
This is actually quite nice, however, I'd rather move the models to a common place first, and then implement annotator there. It seems like you'll need to make the change in 3 places in order to make it consistent at the moment, which is not ideal. 
How far are we in planning to move models/entities in a common repo @feng-tao @jinhyukchang ?

**Comment by feng-tao** [comment]
@verdan, @friendtocephalopods  . Make sense. @jinhyukchang is working with our OSS team to open source that common repo. Currently it is review cycle, will get back to this once it is finished. Stay tuned!

**Comment by jinhyukchang** [comment]
> This is actually quite nice, however, I'd rather move the models to a common place first, and then implement annotator there. It seems like you'll need to make the change in 3 places in order to make it consistent at the moment, which is not ideal.
> How far are we in planning to move models/entities in a common repo @feng-tao @jinhyukchang ?

Good idea @verdan . `amundsen-common` repo just initiated procedure of open sourcing it and I expect it will be done in few weeks but at the same time we don't have control as depends on few external teams within Lyft.

I suggest to adopt this PR and have a separate step to place it in common repo. What do you think?

**Comment by verdan** [comment]
sounds good to me @jinhyukchang 👍 
cc: @feng-tao

File: metadata_service/entity/table_detail.py [hunk 1]
```diff
@@ -1,175 +1,176 @@
...
+        register_as_scheme = True
+
+
+@attr.s(auto_attribs=True, kw_only=True)
+class PiiDetail:
+    pii_semantic_type: Optional[str] = None
+    pii_forget_type: Optional[str] = None
+
 
+class PiiDetailSchema(AttrsSchema):
```
Comment (feng-tao) [inline metadata_service/entity/table_detail.py]: do we use this new model anywhere?

**Pull request merged**